rust/gui-client/src-tauri/src/gui.rs

```diff
@@ -72,6 +76,29 @@ impl GuiIntegration for TauriIntegration {
         } else {
             win.hide().context("Couldn't hide Welcome window")?;
         }
+
+        // Ensure state in frontend is up-to-date.
+        match current_session {
+            Some(session) => self.notify_signed_in(session)?,
+            None => self.notify_signed_out()?,
+        };
```

Comment:
In order to make sure the frontend is definitely up to date, we dispatch an event every time we show/hide it.